src/ios/LiveUpdatePlugin.swift

```diff
@@ -24,44 +24,44 @@
 
      - Returns: A CDVPluginResult object with status and message.
      */
-    func getConfiguration (command: CDVInvokedUrlCommand){
+    @objc func getConfiguration (_ command: CDVInvokedUrlCommand){
         let actionName = "getConfiguration"
         if let configuration = command.arguments[0] as? NSDictionary{
             print("\(actionName): Configuration parameters are: \(configuration)")
 
-            let segmentId = configuration.valueForKey("segmentId") as? String ?? ""
-            let useClientCache = configuration.valueForKey("useClientCache") as? Bool ?? true
-            let params = configuration.valueForKey("params") as? [String:String] ?? [String:String]()
+            let segmentId = configuration.value(forKey: "segmentId") as? String ?? "all"
+            let useClientCache = configuration.value(forKey: "useClientCache") as? Bool ?? true
+            let params = configuration.value(forKey: "params") as? [String:String] ?? [String:String]()
 
-            if (segmentId != ""){
+            if (segmentId != "" && params.isEmpty){
```

Comment:
Why you checking params value here >

Reply:
Instead of segmentId user can give params as the option. So, we need to use "all" segmentId only if segmentId is empty and there are no params provided.

Reply:
AFAIK, based on your changes SegmentId can never be empty as you are keeping it `all` as default segment id. 

Can you tell me how this condition becomes valid ?